vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamDeviceProvider.kt

```diff
@@ -39,26 +43,24 @@ class AdamDeviceProvider(
     private val track: Track,
     private val timer: Timer
 ) : DeviceProvider, CoroutineScope {
-    private val devices: MutableMap<String, AdamAndroidDevice> = ConcurrentHashMap()
+    private val devices: MutableMap<String, ProvidedDevice> = ConcurrentHashMap()
     private val logger = MarathonLogging.logger("AdamDeviceProvider")
 
     private val channel: Channel<DeviceProvider.DeviceEvent> = unboundedChannel()
-    private val bootWaitContext =
-        newFixedThreadPoolContext(androidConfiguration.threadingConfiguration.bootWaitingThreads, "BootWaitThreadPool")
-    override val coroutineContext: CoroutineContext
-        get() = bootWaitContext
-
-    private val adbCommunicationContext: CoroutineContext by lazy {
-        newFixedThreadPoolContext(androidConfiguration.threadingConfiguration.adbIoThreads, "AdbIOThreadPool")
-    }
+    override val coroutineContext: CoroutineContext by lazy { newFixedThreadPoolContext(1, "DeviceMonitor") }
+    private val adbCommunicationContext: CoroutineContext by lazy { newFixedThreadPoolContext(androidConfiguration.threadingConfiguration.adbIoThreads, "AdbIOThreadPool") }
```

Comment:
Do we need this fixedThreadPoolContext? Can we share some common dispatcher and create context on top of it?